In [31]:
import pandas as pd
import numpy as np
import nltk
import re
import string
import matplotlib.pyplot as plt
import numpy as np
%matplotlib inline

from sklearn.cross_validation import train_test_split
from sklearn.metrics import confusion_matrix
from sklearn import metrics
from nltk.stem.wordnet import WordNetLemmatizer
import warnings

from sklearn.neural_network import MLPClassifier
from sklearn.metrics import roc_auc_score
from sklearn.linear_model import SGDClassifier

from scipy.stats.stats import pearsonr

import numpy as np
import pandas as pd
from sklearn.cross_validation import KFold
from sklearn.svm import SVR
from sklearn.linear_model import LinearRegression
from sklearn.ensemble import RandomForestRegressor

ESTIMATORS = {
    "linear": LinearRegression(),
    "svr":  SVR(C=1.0, epsilon=0.2),
    "rfr": RandomForestRegressor(n_estimators=250),
}

In [57]:
temp_values = ['MSRpar', 'MSRvid', 'SMTeuroparl', 'surprise.OnWN',
       'surprise.SMTnews', 'headlines', 'OnWN', 'FNWN', 'deft-forum',
       'deft-news', 'tweet-news', 'answers-forums', 'answers-students',
       'belief', 'images', 'answer-answer', 'plagiarism',
       'question-question', 'postediting']

In [58]:
for val in temp_values:
    train_modified = pd.read_csv('final-modified.csv')
    train_data1 = train_modified[train_modified['type']==val]
    train_data1_labels = train_data1['original_score'].values
    train_data1 = train_data1.drop(['Unnamed: 0', 'year', 'type', 'sentence1', 'sentence2', 'cleaned_S1','cleaned_S2', 'cleaned_tagged_S1', 'cleaned_tagged_S2','original_score'],axis=1)
    train_data1 = train_data1.fillna(0)

    n_folds = 10
    train_predictions = pd.DataFrame({'id': range(train_data1.shape[0])})
    train_predictions.set_index(['id'])
    for name, estimator in ESTIMATORS.items():
        #print("Model: ", name)

        kf = KFold(train_data1.shape[0], n_folds=n_folds)
        hold_prediction = []
        eval_score = 0

        for i, (train_index, test_index) in enumerate(kf):
            X_train, X_val = train_data1.iloc[train_index], train_data1.iloc[test_index]
            y_train, y_val = train_data1_labels[train_index], train_data1_labels[test_index]
            estimator.fit(X_train, y_train)
            hold_prediction.extend(estimator.predict(X_val))
            #print("Finished fold",i)
        train_predictions[name] = hold_prediction
    
    print("Statistics for algorithm:",val)
    for col in ESTIMATORS.keys():
        print(col,":",pearsonr(train_predictions[col], train_data1_labels)[0])

Statistics for algorithm: MSRpar
rfr : 0.601822754905
svr : 0.595825498992
linear : 0.603218013389
Statistics for algorithm: MSRvid
rfr : 0.835618361148
svr : 0.816021724352
linear : 0.817675281973
Statistics for algorithm: SMTeuroparl
rfr : 0.682188937363
svr : 0.546935592744
linear : 0.508756418046
Statistics for algorithm: surprise.OnWN
rfr : 0.687808392775
svr : 0.694557394755
linear : 0.706064909799
Statistics for algorithm: surprise.SMTnews
rfr : 0.703835366926
svr : 0.529104156873
linear : 0.573635168224
Statistics for algorithm: headlines
rfr : 0.740310171788
svr : 0.746410419625
linear : 0.742542805858
Statistics for algorithm: OnWN
rfr : 0.837744411491
svr : 0.783871157708
linear : 0.792979998812
Statistics for algorithm: FNWN
rfr : 0.297620307694
svr : 0.398700189823
linear : 0.419273323689
Statistics for algorithm: deft-forum
rfr : 0.552775725655
svr : 0.501178488807
linear : 0.481079610026
Statistics for algorithm: deft-news
rfr : 0.702813824483
svr : 0.719635634775
linear

### Statistics for algorithm: MSRpar
rfr : 0.601822754905
svr : 0.595825498992
linear : 0.603218013389
### Statistics for algorithm: MSRvid
rfr : 0.835618361148
svr : 0.816021724352
linear : 0.817675281973
### Statistics for algorithm: SMTeuroparl
rfr : 0.682188937363
svr : 0.546935592744
linear : 0.508756418046
### Statistics for algorithm: surprise.OnWN
rfr : 0.687808392775
svr : 0.694557394755
linear : 0.706064909799
### Statistics for algorithm: surprise.SMTnews
rfr : 0.703835366926
svr : 0.529104156873
linear : 0.573635168224
### Statistics for algorithm: headlines
rfr : 0.740310171788
svr : 0.746410419625
linear : 0.742542805858
### Statistics for algorithm: OnWN
rfr : 0.837744411491
svr : 0.783871157708
linear : 0.792979998812
### Statistics for algorithm: FNWN
rfr : 0.297620307694
svr : 0.398700189823
linear : 0.419273323689
### Statistics for algorithm: deft-forum
rfr : 0.552775725655
svr : 0.501178488807
linear : 0.481079610026
### Statistics for algorithm: deft-news
rfr : 0.702813824483
svr : 0.719635634775
linear : 0.725616984897
### Statistics for algorithm: tweet-news
rfr : 0.778849841396
svr : 0.795841753563
linear : 0.794089919529
### Statistics for algorithm: answers-forums
rfr : 0.679890213827
svr : 0.71651564492
linear : 0.717706895019
### Statistics for algorithm: answers-students
rfr : 0.759864856717
svr : 0.749711038533
linear : 0.749288351273
### Statistics for algorithm: belief
rfr : 0.715147975564
svr : 0.7474587939
linear : 0.743796223751
### Statistics for algorithm: images
rfr : 0.824199901462
svr : 0.834070586541
linear : 0.836810451738
### Statistics for algorithm: answer-answer
rfr : 0.456653013023
svr : 0.493954569536
linear : 0.48275129827
### Statistics for algorithm: plagiarism
rfr : 0.806547374988
svr : 0.807327497773
linear : 0.81470858138
### Statistics for algorithm: question-question
rfr : 0.599103434565
svr : 0.644245387092
linear : 0.656265243403
### Statistics for algorithm: postediting
rfr : 0.825802506921
svr : 0.844453433786
linear : 0.838484463606